### PR TITLE
DAOS-9087 doc: doc OFI interface/domain

### PR DESCRIPTION
**Pull request by wangshilong**
After upgrading latest master, one of site reported daos
client could not start any more, it turned out they only
export OFI_INTERACE without OFI_DOMAIN, after commit:

004bcec5ab910611690dbd15210605e8134d94e4

we will assume we don't need OFI_DOMAIN, output warning
message and document it explictly.

doc-only: true
Signed-off-by: Wang Shilong <shilong.wang@intel.com>

### CODE REVIEW

File: src/mgmt/cli_mgmt.c [hunk 1]
```diff
@@ -403,6 +403,8 @@ int dc_mgmt_net_cfg(const char *name)
 		/* If the client env didn't provide a domain, we can assume we don't need one. */
 		if (ofi_domain)
 			D_INFO("Using client provided OFI_DOMAIN: %s\n", ofi_domain);
+		else
+			D_WARN("Client didn't provide OFI_DOMAIN, assuming we don't need one");
```
Comment (kjacque) [inline src/mgmt/cli_mgmt.c:407]: I'm not sure about logging this message at WARN level. This is not an error condition unless they are using a provider that requires a domain (like verbs). You could print something at INFO level, for example, "Using client-provided OFI_INTERFACE without OFI_DOMAIN," but I'm not sure how useful it is, given all the env variables are printed at INFO level anyway.

Comment (frostedcmos) [reply src/mgmt/cli_mgmt.c:407]: if client does not provide ofi_domain then cart automatically assumes  ofi_domain == ofi_interface. this works for all providers, except for verbs and cxi, where domains and interface names do not match. this is why those providers require additional setting. 

It probably would be better to warn in the cart layer at init time, if (provider == cxi or verbs_rxm) and ofi_domain is not specified then print the warning there

Comment (wangshilong) [reply src/mgmt/cli_mgmt.c:407]: Makes sense.

**Review by kjacque — APPROVED** [review]

File: src/cart/crt_init.c [hunk 2]
```diff
@@ -795,6 +795,10 @@ int crt_na_ofi_config_init(int provider)
 	if (domain == NULL) {
 		D_DEBUG(DB_ALL, "OFI_DOMAIN is not set. Setting it to %s\n",
 			interface);
+		if (provider == CRT_NA_OFI_VERBS_RXM ||
+		    provider == CRT_NA_OFI_CXI)
+			D_WARN("Domain and interface name expect to be different "
```
Comment (frostedcmos) [inline src/cart/crt_init.c:800]: this probably should say 'expected'

**Review by frostedcmos — APPROVED** [review]

**Review by kjacque — APPROVED** [review]

**Review by frostedcmos — APPROVED** [review]

**Pull request merged**